x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -166,6 +170,26 @@ export function App({
     });
   }, [onAppLeave, lastKnownDoc, isSaveable, persistedDoc, application.capabilities.visualize.save]);
 
+  const getLegacyUrlConflictCallout = useCallback(() => {
+    // This function returns a callout component *if* we have encountered a "legacy URL conflict" scenario
+    if (spaces && sharingSavedObjectProps?.outcome === 'conflict' && persistedDoc?.savedObjectId) {
+      // We have resolved to one object, but another object has a legacy URL alias associated with this ID/page. We should display a
+      // callout with a warning for the user, and provide a way for them to navigate to the other object.
+      const currentObjectId = persistedDoc.savedObjectId;
+      const otherObjectId = sharingSavedObjectProps?.aliasTargetId!; // This is always defined if outcome === 'conflict'
+      const otherObjectPath = http.basePath.prepend(getFullPath(otherObjectId));
```

Comment:
When I loaded in the test data, I navigated to `http://localhost:5601/s/test3/app/lens#/edit/15522780-0a80-11ec-a4bf-616628b395be-test3?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))` and clicked the button to "Go to other Lens visualization".
I expected to be redirected to: `http://localhost:5601/s/test3/app/lens#/edit/15522780-0a80-11ec-a4bf-616628b395be-test3-newId?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))` (it _should_ just change the object ID)
Instead, I was redirected to: `http://localhost:5601/s/test3/app/lens/s/test3/app/lens#/edit/15522780-0a80-11ec-a4bf-616628b395be-test3-newId`

So there are a couple issues:
1. The space prefix `/s/test3` and app path `/app/lens` are repeated (somehow the lens still works though?!)
2. The global filter is stripped off

This is partially my fault (sorry!), as the POC uses `http.basePath` to construct the URL; that was some old code that should have been updated to match the new implementation of `getLegacyUrlConflict()`.

Looking at the functions in `lens/common`, I think you need to change it to something like this
```suggestion
      const otherObjectPath = getEditPath(otherObjectId /*, timeRange (not sure where to get this? */);
```

If you can provide `timeRange` I assume that it should keep the global filter intact. I'm not familiar with the global filter, though; are there other fields to be concerned about besides the time range?
Also, I tried `initialInput?.timeRange` but that didn't seem to work, as `initialInput` is undefined.

Reply:
I used `history` to construct the path, we do it in some other places so I think it's fine. Thank you for spotting these issues!